contrib/layout.py

```diff
@@ -0,0 +1,115 @@
+#
+# Fibonacci dwindle layout for river, somewhat similar to bspwm, implemented in
+# simple pthon. Reading this code should help you get a basic understanding of
+# how to use river-layout to create a basic layout generator.
+#
+# This depends on pywayland: https://github.com/flacjacket/pywayland/
+#
+# Q: Wow, this looks complicated!
+# A: For simple layouts, you really only need to care about what's in the
+#    layout_handle_layout_demand() function. And the rest isn't as complicated
+#    as it looks.
+
+import mmap
+import time
+from pywayland.client import Display
+from pywayland.protocol.wayland import WlOutput
+from pywayland.protocol.river_layout_v3 import RiverLayoutManagerV3
+
+layout_manager = None
+outputs = []
+
+def layout_handle_layout_demand(layout, view_count, usable_w, usable_h, tags, serial):
+    x = 0
+    y = 0
+    w = usable_w
+    h = usable_h
+    i = 0
+    while i < view_count:
+        if i == view_count - 1:
+            layout.push_view_dimensions(x, y, w, h, serial)
+        else:
+            if i % 2 == 0:
+                w = int(w/2)
+                layout.push_view_dimensions(x, y, w, h, serial)
+                x += w
+            else:
+                h = int(h/2)
+                layout.push_view_dimensions(x, y, w, h, serial)
+                y += h
+        i += 1
+
+    # Committing the layout means telling the server that your code is done
+    # laying out windows. Make sure you have pushed exactly the right amount of
+    # view dimensions, a mismatch is a fatal protocol error.
+    # 
+    # You also have to provide a layout name. This is a user facing string that
+    # the server can forward to status bars. You can use it to tell the user
+    # which layout is currently in use. You could also add some status
+    # information status information about your layout, which is what we do here.
+    layout.commit(str(view_count) + " windows layout out by python", serial)
+
+class Output(object):
+    def __init__(self):
+        self.output = None
+        self.layout = None
+        self.id = None
+
+    def destroy(self):
+        if self.layout != None:
```

Comment:
Although it might work most of the time, it's a bad practice to use `==` checking if something's `None`. You should use `self.layout is None` (or `is not None` in some cases below). The reason for that is `==` can be technically overridden for some types and not behave the way you expect, while `is None` always works as expected.

Note there are several of these comparisons below.